### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by davidbrochart**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+env:
+  PYTEST_ADDOPTS: "--color=yes"
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```
Comment (johnslavik) [inline .github/workflows/test.yml:20]: I find this matrix slightly suboptimal.
https://blog.jaraco.com/efficient-use-of-ci-resources

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4.1.1
+      - name: Install Poetry
+        run: pipx install poetry==1.8.2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4.7.1
```
Comment (johnslavik) [inline .github/workflows/test.yml:29]: I think it is worth setting to setup-python@v5.
[v5.1.0 is the latest release](https://github.com/actions/setup-python/releases).

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    branches:
+    - main
+
+env:
+  PYTEST_ADDOPTS: "--color=yes"
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
```
Comment (johnslavik) [inline .github/workflows/test.yml:17]:
```suggestion
    strategy:
      fail-fast: false
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4.1.1
+      - name: Install Poetry
+        run: pipx install poetry==1.8.2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4.7.1
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'poetry'
```
Comment (johnslavik) [inline .github/workflows/test.yml:32]:
```suggestion
          cache: poetry
```
For consistency.

**Review by johnslavik — CHANGES_REQUESTED** [review]

**Comment by willmcgugan** [comment]
Thanks @davidbrochart 

Merging for now. @bswck feel free to PR your changes. I don't know enough about GA to review.

**Pull request merged**